src/controllers/zodiac.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import {
+  GetLastElementJSNatFromSignData,
+  getTodayAllSignRecordsFromDB,
+} from "../dal/zodiacdeep";
+import { ISignRecord } from "../interfaces/signs";
+import { getText } from "../helpers/arrayProcessing";
+
+export async function getZodiac(req: Request, res: Response): Promise<void> {
+  {
+    const sign = req?.query?.sign;
+    let totalResult = "";
+
+    if (typeof sign !== "undefined") {
```

Comment:
Here you can reduce the if-else nesting by combining both cases and returning the response instead of assigning it.